### PR TITLE
Suppress sidenotes when we fail to find nonempty footnote contents

### PR DESCRIPTION
**Pull request by jimrandomh**
This comes up in two known situations:
 * On some old posts authored in raw HTML, where we aren't able to find the footnote body, such as [this one](https://www.lesswrong.com/posts/ACGeaAk6KButv2xwQ/the-halo-effect)
 * On comments that are truncated in a way that removes the footnote contents

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1208218877745086) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/linkPreview/FootnotePreview.tsx [hunk 1]
```diff
@@ -132,10 +132,25 @@ const FootnotePreview = ({classes, href, id, rel, children}: {
...
+    // things. Eg https://www.lesswrong.com/posts/ACGeaAk6KButv2xwQ/the-halo-effect
+    // We can't just condition on the footnote containing non-whitespace text,
+    // because footnotes sometimes have their number and backlink in a place that
+    // would be mistaken for their body. Our current heuristic is that a footnote
+    // is nonempty if it contains at least one <p> which contains non-whitespace
+    // text, which might false-negative on rare cases like an image-only footnote
+    // but which seems to work in practice.
+    footnoteContentsNonempty = !!footnoteContentsElement
+      && !!Array.from(footnoteContentsElement.querySelectorAll("p"))
+        .reduce((acc, p) => acc + p.textContent, "").trim();
```
Comment (Discordius) [inline packages/lesswrong/components/linkPreview/FootnotePreview.tsx:153]: I feel like this reduce could be an `a.some`

Comment (b0b3rt) [reply packages/lesswrong/components/linkPreview/FootnotePreview.tsx:153]: `.some(p => !!p.textContent.trim())`?

**Review by s-cheng — COMMENTED** [review]
No objections from EAF

File: packages/lesswrong/components/linkPreview/FootnotePreview.tsx (hunk 1 above)
Comment (jimrandomh) [reply packages/lesswrong/components/linkPreview/FootnotePreview.tsx:153]: (Agreed that these are equivalent and simpler; this line was preexisting and I only reformatted it without changing that part.)

**Pull request merged**